modules/sdk-coin-xrp/src/ripple.ts

```diff
@@ -32,10 +31,10 @@ const signWithPrivateKey = function (txHex, privateKey, options) {
 
   let tx;
   try {
-    tx = binary.decode(txHex);
+    tx = JSON.parse(txHex);
   } catch (e) {
+    tx = binary.decode(txHex);
     try {
```

Comment:
I suppose you meant to have this inside the below try block?
```suggestion
    try {
     tx = binary.decode(txHex);
```